### PR TITLE
Add examples of storage-opts and log-opts for the daemon

### LINKED ISSUE #967: Add examples of storage-opts and log-opts for the daemon
File: [engine/reference/commandline/dockerd.md](https://docs.docker.com/engine/reference/commandline/dockerd/)

There are no examples of how to add storage-opts or log-opts, nor anything in second-level lists of properties, really, to /etc/docker/daemon.json.  Here's an example that would save a few folks' time.

```
{
  "storage-driver": "devicemapper",
  "storage-opts": [
    "dm.thinpooldev=/dev/mapper/vg0-pool0",
    "dm.use_deferred_deletion=true",
    "dm.use_deferred_removal=true"
  ],
  "log-driver": "json-file",
  "log-opts": {
    "max-size": "10m",
    "max-file": "10"
  }
}
```

### PR DESCRIPTION
**Pull request by alvin319**
Signed-off-by: Alvin Deng <alvin.q.deng@utexas.edu>

**- What I did**
Added example JSON for the device mapper documentation

Fixes docker/docker.github.io#967

### CODE REVIEW

**Review by mdlinville — APPROVED** [review]
LGTM

File: docs/reference/commandline/dockerd.md [hunk 1]
```diff
@@ -302,6 +302,24 @@ options for `zfs` start with `zfs` and options for `btrfs` start with `btrfs`.
...
+
+```json
+{
+  "storage-driver": "devicemapper",
+  "storage-opts": [
+    "dm.thinpooldev=/dev/mapper/thin-pool",
+    "dm.use_deferred_deletion=true",
+    "dm.use_deferred_removal=true"
+  ],
+  "log-driver": "json-file",
```
Comment (cpuguy83) [inline docs/reference/commandline/dockerd.md]: The line above this json block says this is an example for devicemapper options, seems weird to add log driver stuff here.

**Review by cpuguy83 — DISMISSED** [review]

**Review by alvin319 — COMMENTED** [review]
That makes sense!

**Review by vdemeester — APPROVED** [review]
LGTM 👼

**Review by mdlinville — APPROVED** [review]

**Comment by mstanleyjones** [comment]
Thanks @alvin319 !

**Pull request merged**